src/Sarif/Baseline/ResultMatching/DataStructures/ExtractedResult.cs

```diff
@@ -15,7 +12,65 @@ namespace Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching
     public class ExtractedResult
     {
         public Result Result { get; set; }
-        
         public Run OriginalRun { get; set; }
+        public string RuleId { get; set; }
+
+        public ExtractedResult(Result result, Run run)
+        {
+            Result = result;
+            OriginalRun = run;
+
+            // Look up and cache the RuleId
+            RuleId = result.ResolvedRuleId(run);
+        }
+
+        /// <summary>
+        ///  Match the 'Category' of two ExtractedResults (Tool and RuleId)
+        /// </summary>
+        /// <param name="other">ExtractedResult to match</param>
+        /// <returns>True if Category is identical, False otherwise</returns>
+        public bool MatchesCategory(ExtractedResult other)
+        {
+            return this.RuleId == other.RuleId;
+
+            // Tool (checked on Runs beforehand)
+            // Consider: CorrelationGuid, Level, Rank?
+        }
+
+        /// <summary>
+        ///  Match the 'What' properties of two ExtractedResults (Fingerprint and Snippet)
+        /// </summary>
+        /// <param name="other">ExtractedResult to match</param>
+        /// <returns>True if *any* 'What' property matches, False otherwise</returns>
+        public bool MatchesWhat(ExtractedResult other)
```

Comment:
>MatchesWhat [](start = 20, length = 11)

It would be nice if the name of MatchesWhat and MatchesWhere indicated if it used 'any' or 'all' semantics.

Reply:
Good idea - renamed to "MatchesAllWhere" and "MatchesAnyWhat".